### PR TITLE
added receipt and state verification

### PR DESCRIPTION
**Pull request by willeslau**
## Changes

- Added verification for state and receipt

## Issues

- #20

### CODE REVIEW

File: filecoindot/src/lib.rs [hunk 1]
```diff
@@ -64,6 +65,12 @@ pub mod pallet {
         type WeightInfo: WeightInfo;
         /// The timeout of the http requests of ocw in milliseconds
         type OffchainWorkerTimeout: Get<u64>;
+        type AMTNode: filecoindot_proofs::AMTNode;
+        type StateKey: Eq;
+        type StateVal: Eq;
+        type StateHash: filecoindot_proofs::HashedBits;
+        type HAMTNode: filecoindot_proofs::HAMTNode<Self::StateKey, Self::StateVal, Self::StateHash>;
+        type Verify: filecoindot_proofs::Verify;
```
Comment (mattsse) [inline filecoindot/src/lib.rs:73]: are those necessary, can't we just call into the crate directly, or are there some generic restrictions in `proofcrate::verify`?

Comment (willeslau) [reply filecoindot/src/lib.rs:73]: yeah, I changed this already

**Review by mattsse — APPROVED** [review]
lgtm, but needs rebase

File: filecoindot-proofs/src/verify.rs [hunk 2]
```diff
@@ -44,3 +30,21 @@ impl ProofVerify {
...
+    /// Verify the proof and the the trie actually matches. Each cid in the proof
+    /// is connected to its neighbours. The proof should match exactly in path from
+    /// the root to the node.
+    /// Note that proof[proof.len-1] == root_cid_bytes. This function does not assume
+    /// the head of the proof to be equal to node_cid, as long as it's in the proof.
+    fn verify_proof<N>(proof: Vec<Vec<u8>>, node_cid: Vec<u8>) -> Result<(), Error>
+    where
+        N: GetCid + for<'de> serde::Deserialize<'de>,
+    {
+        let node_cid = Cid::try_from(node_cid).map_err(|_| Error::VerificationFailed)?;
```
Comment (ivanceras) [inline filecoindot-proofs/src/verify.rs:44]: I think this should be a new variant `Error::CidError(cid::Error)` to just forward the cid parsing error.

Comment (ivanceras) [reply filecoindot-proofs/src/verify.rs:44]: oh wait, nvm, this is a pallet error which doesn't need to be that detailed to save up memory space.

**Review by ivanceras — COMMENTED** [review]
This is just a post-review of a merged PR to get used to the process of reviewing more PRs.

**Pull request merged**